Add EIP: Ground Zero (#7488)
ethereum/EIPs: EIPS/eip-0.md

---

> eip: 0

Reviewer: ```suggestion
eip: 7488
```

title: Ground Zero
author: ProphetXZ <138210332+ProphetZX@users.noreply.github.com> [ProphetZX](https://github.com/ProphetZX)
type: Core
status: Draft
created: 2023-06-30
---

## Simple Summary

This EIP "Ground Zero" suggests to switch mainnet to permissioned to lower gas prices and transaction costs. The adjustment requires protcol and governance changes. 

## Abstract

The EIP idea is to brake the IPFS and Swarm store and broadcasting and instead suggests scalling down ETHEREUM ressources, so that the community may be served in a more sustainable way. In addition, to mitigate the currently existing problem with increasingly high gas prices, ETHEREUM would benefit from switching to a permissioned mainnet (after successfully deploying a testnet version). This would greatly lower the number of requests and decrease the load for operating nodes. A further improvement would be to remove legacy APIs, so nodes no longer answer unnecessary public API requests. 

> ## Rationale

Reviewer: You'll need to follow the header order from EIP-1.

> Since its humble beginnings, the Ethereum project has been open to the public and seen many contributions. The past years, however, have made it clear that such a p.p. open ecosystem design leads to a wide array of problems such as ongoing consensus problems, increasingly high transaction costs, cases of fraud, false ownership claims of assets such as the [DAO hack and resulting fork](https://www.coindesk.com/learn/understanding-the-dao-attack/), illegal contracts and privacy intrusion, data theft, artifical and malicious content, and many more such things. 

Reviewer: Please expand abbreviations ("p.p.") the first time they are used.

> Since its humble beginnings, the Ethereum project has been open to the public and seen many contributions. The past years, however, have made it clear that such a p.p. open ecosystem design leads to a wide array of problems such as ongoing consensus problems, increasingly high transaction costs, cases of fraud, false ownership claims of assets such as the [DAO hack and resulting fork](https://www.coindesk.com/learn/understanding-the-dao-attack/), illegal contracts and privacy intrusion, data theft, artifical and malicious content, and many more such things. 

Reviewer: ```suggestion
Since its humble beginnings, the Ethereum project has been open to the public and seen many contributions. The past years, however, have made it clear that such a p.p. open ecosystem design leads to a wide array of problems such as ongoing consensus problems, increasingly high transaction costs, cases of fraud, false ownership claims of assets such as the DAO hack and resulting fork (see [EIP-779](./eip-779.md)), illegal contracts and privacy intrusion, data theft, artificial and malicious content, and many more such things. 
```

> Halting the IPFS and Swarm storage process, which enabled and multiplied above problems, as well as historically drained the ETHEREUM ecosystem and infrastructure of important ressources and instead developing use cases for the original community would be a remedy for almost all of these problems.

Reviewer: Please change "ETHEREUM" to "Ethereum" in the entire document.

```suggestion
Halting the IPFS and Swarm storage process, which enabled and multiplied above problems, as well as historically drained the Ethereum ecosystem and infrastructure of important resources and instead developing use cases for the original community would be a remedy for almost all of these problems.
```

## Specification

The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.

> Mainnet (ID: 1) MUST be switched to permissioned. In part, this may be achieved by halting IPFS, Swarm, and Whisper, after a successfull testnet version has been deployed. This will affect existing DApps, but almost none of these have been used in the past by the community anyway. So the impact of this protocol change should be minor. For active transactions and smart contracts, opcode 0x00 SHOULD be executed. As a necessary and sufficient condition, opcode 0xff for the zero address SHALL be exectued as well.

Reviewer: > Mainnet (ID: 1) MUST be switched to permissioned.

This does not provide enough detail. What does a permissioned Ethereum look like? Are you talking about [clique](https://eips.ethereum.org/EIPS/eip-225)? Something else? To be an EIP, a proposal needs to be specified to a sufficient level of detail that it can be implemented.

> This will affect existing DApps, but almost none of these have been used in the past by the community anyway.

This belongs in the backwards compatibility or rationale sections.

> For active transactions and smart contracts, opcode 0x00 SHOULD be executed. As a necessary and sufficient condition, opcode 0xff for the zero address SHALL be exectued as well.

Could you clarify this a bit more in the document? Do you mean that instead of the currently deployed bytecode, these opcodes will be executed instead?

> Potentially, an adjustment to devp2p, EIP-8, would be required as well and Oracle connections should be switched.

Reviewer: You'll need to describe the exact changes you want made.
